### PR TITLE
Generic alert message for pending List

### PR DESCRIPTION
**Pull request by jenniemc**
## Summary
When any of the systems are unavailable, the frontend will display a generic message to the user that their appointment list may be unavailable to view.
- Display a generic message to the user that their request appointment list may be unavailable to view.
- The fetch error codes will differentiate between request/booked failures. There is also a fallback in case we get in to a scenario where the failure wasn't due to a fetching of requests or booked appointments. The error codes codes are as follows:
10000 - Failure to fetch from VSP (general fallback)
10005 - Failure to fetch Booked Appointments
10006 - Failure to fetch Requests

- Example of a response that contains various backend service failures
```
"failures": [
            {
                "system": "VSP",
                "status": "500",
                "code": 10000,
                "message": "Could not fetch appointments from Vista Scheduling Provider",
                "detail": "icn=500104V500104 startDate=2016-01-25T00:00Z endDate=2025-12-26T00:00Z"
            },
            {
                "system": "VSP",
                "status": "500",
                "code": 10005,
                "message": "Could not fetch booked appointments from Vista Scheduling Provider",
                "detail": "icn=500104V500104 startDate=2016-01-25T00:00Z endDate=2025-12-26T00:00Z"
            },
            {
                "system": "VSP",
                "status": "500",
                "code": 10006,
                "message": "Could not fetch appointment requests from Vista Scheduling Provider",
                "detail": "icn=500104V500104 startDate=2016-01-25T00:00Z endDate=2025-12-26T00:00Z"
            }
        ]
```

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#55279

## Testing done

To test locally:

1. In the mock file, uncomment line 38 and comment out line 41. This scenario will test the error code 10000 where it is a general error fetch from VSP. It will display "We can't display all your requests." in the pending list page.

| Code | Before | After |
| --- | --- | --- |
| 10000 |Screenshot 2023-04-17 at 2 27 32 PM | Screenshot 2023-04-17 at 2 06 59 PM|

2. In the mock file, uncomment line 38 and comment out line 41. In the `meta_failures.json` remove lines 16-22 and lines 30-36. This scenario will test the error code 10005 where it does not fetch booked appointments. Therefore it has no impact to pending appointment list display.

| Code | Before | After |
| --- | --- | --- |
| 10005 |Screenshot 2023-04-17 at 2 27 32 PM | Screenshot 2023-04-17 at 2 27 32 PM |

3. In the mock file, uncomment line 38 and comment out line 41. In the `meta_failures.json`  remove lines 16-39. This scenario will test the error code 10006 where it does not fetch request appointments. Therefore will display the error message "We can't display all your requests."  in pending list. However, it has no impact to upcoming and past list.

| Code | Before | After |
| --- | --- | --- |
| 10006 |Screenshot 2023-04-17 at 2 27 32 PM | Screenshot 2023-04-17 at 2 06 59 PM|

## Screenshots
See above

## What areas of the site does it impact?
VAOS appointment list

## Acceptance criteria

- [ ]  Display the message to match the  doc copy https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/content/alert-copy-docs/upcoming-view.md#pending-appointments

- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx [hunk 1]
```diff
@@ -1,38 +1,74 @@
...
 import { selectBackendServiceFailuresInfo } from '../redux/selectors';
-import { FETCH_STATUS } from '../../utils/constants';
+import { FETCH_STATUS, ERROR_CODES } from '../../utils/constants';
+
+function displayType(errorCodes, location) {
+  const isPending = location.pathname.endsWith('/pending');
+  const isPast = location.pathname.endsWith('/past');
+  const isUpcoming = location.pathname.endsWith('/');
+
+  if (errorCodes.some(code => code === 10000 || code === 10006) && isPending) {
```
Comment (ryanshaw) [inline src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: Are we supposed to look for specific error codes in the response? I was under the impression that we should display this for any error that comes through? @simiadebowale, can you clarify?

Comment (jenniemc) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: > Are we supposed to look for specific error codes in the response? I was under the impression that we should display this for any error that comes through? @simiadebowale, can you clarify?

I thought the generic message was regardless of which system is down: VSP, VVS and/or HSRM; display the message if any one of those systems experience failure retrieving booked appointment, request appointment or both.

Comment (ryanshaw) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: Yes, I believe that is correct, I was a bit unclear I guess. I meant the specific error codes (10006, 10000, etc). I wasn't sure if those would be entirely consistent or if we even knew all the possible error codes, that is why I originally was just checking for the length of that array. 

If we know all the error codes, then this is great and more precise.

Comment (jenniemc) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: I guess that's why we need @simiadebowale get some clarification 😉

Comment (simiadebowale) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: @ryanshaw - we're not using code/system to deffrentiate the system returning errors but we want to differentiate between errors fetching pending appts and booked appts

Comment (ryanshaw) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: @simiadebowale Gotcha. That makes sense.

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: I don't know if this matters right now but the error messages are the same for all the error codes.

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: Is the goal to use the error message returned from the api call?

Comment (simiadebowale) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: @vbahinwillit - we're still displaying the generic error message. There are cases where BE systems aren't down for fetching upcoming/past appts but are down for fetching Pending appts. In this case, we want to only display this generic error message when the user is viewing their pending appts list and not when they're viewing their upcoming/past appts list

**Review by simiadebowale — APPROVED** [review]

**Comment by simiadebowale** [comment]
@vbahinwillit - we're still displaying the generic error message. There are cases where BE systems aren't down for fetching upcoming/past appts but are down for fetching Pending appts. In this case, we want to only display this generic error message when the user is viewing their pending appts list and not when they're viewing their upcoming/past appts list

File: src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: Still confused. What error codes are returned from the BE for the different scenarios? Maybe that might help my understanding.

Comment (simiadebowale) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: sure @vbahinwillit 
New codes:
10000 - Failure to fetch from VSP (Generic Error)
10005 - Failure to fetch Booked Appointments
10006 - Failure to fetch Requests

So in this case, if we get an error code of 10006, we assume there's only an error fetching requests so we only display the error message in the pending appt list view

Comment (simiadebowale) [reply src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx:12]: There's more info here. I was sent [this link] as well (https://coderepo.mobilehealth.va.gov/projects/EAS/repos/mobile-appointment-service/browse/mobile-appointment-service/src/main/java/gov/va/mobile/appointment/v2/exceptions/ErrorCode.java#67) but I don't have access so I can't speak to the contents of that page.

**Pull request merged**